src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -552,16 +548,10 @@ def _forward(self, model_inputs, generate_kwargs=None):
                 **generate_kwargs,
             )
             out = {"tokens": tokens}
-        elif self.type == "seq2seq_whisper":
-            stride = model_inputs.pop("stride", None)
-            tokens = self.model.generate(
-                input_features=model_inputs.pop("input_features"),
-                logits_processor=[WhisperTimeStampLogitsProcessor()] if return_timestamps else None,
-                **generate_kwargs,
-            )
-            out = {"tokens": tokens}
-            if stride is not None:
-                out["stride"] = stride
+            if self.type == "seq2seq_whisper":
```

Comment:
We needed to `pop`  before generate.

If there's no need for it to pop before we can simplify be simply setting something like:

`out = {**out, **model_inputs}`  or something slightly along those lines including only `stride`.